### PR TITLE
Attribute list delimiter option. Fixes #98

### LINKED ISSUE #98: Attribute list delimiter option
Hi!

Is it possible to add something similar to slim's [Attribute list delimiter option `:attr_list_delims`](https://github.com/slim-template/slim#available-options)? It would resolve a conflict between the interpolation markers `{{ }}` some js libraries use and slime's own attribute list delimiters `{ }`.

### PR DESCRIPTION
**Pull request by bozydar**

### CODE REVIEW

File: lib/slime/parser.ex [hunk 1]
```diff
@@ -15,10 +15,19 @@ defmodule Slime.Parser do
   @smart    "="
 
   @attr_delim_regex ~r/[ ]+(?=([^"]*"[^"]*")*[^"]*$)/
+  @attr_list_delims Application.get_env(:slime, :attr_list_delims, %{"[" => "]", "(" => ")"})
```
Comment (Rakoth) [inline lib/slime/parser.ex]: We should support `{}` by default.

File: lib/slime/parser.ex [hunk 2]
```diff
@@ -129,15 +138,19 @@ defmodule Slime.Parser do
...
+
   defp parse_attributes(line) do
-    match = @attr_group_regex |> Regex.run(line) |> List.first
-    offset = String.length(match)
-    {attrs, rem} = line |> String.split_at(offset)
-    attrs = parse_attributes(attrs, [])
-    {rem, attrs}
+    delim = String.first(line)
+    if Dict.has_key?(@attr_list_delims , delim) do
+      line = String.replace_prefix(line, delim, "")
```
Comment (Rakoth) [inline lib/slime/parser.ex]: `String.replace_prefix/3` is a Elixir 1.2 feature,  so the current travis build is failing. Please fix it first.

File: lib/slime/parser.ex [hunk 3]
```diff
@@ -129,15 +138,19 @@ defmodule Slime.Parser do
...
-  defp parse_attributes("{" <> line), do: parse_wrapped_attributes(line, "}")
+
   defp parse_attributes(line) do
-    match = @attr_group_regex |> Regex.run(line) |> List.first
-    offset = String.length(match)
-    {attrs, rem} = line |> String.split_at(offset)
-    attrs = parse_attributes(attrs, [])
-    {rem, attrs}
+    delim = String.first(line)
+    if Dict.has_key?(@attr_list_delims , delim) do
```
Comment (Rakoth) [inline lib/slime/parser.ex:145]: May be we should provide better error message for case when closing delim is not what we are expecting, for example `div[id="test"}` will result in match error in `parse_wrapped_attributes` as far as I see.

**Comment by Rakoth** [comment]
Thanks for your contribution, @bozydar! I left some comments above.

File: lib/slime/parser.ex [hunk 4]
```diff
@@ -142,7 +142,7 @@ defmodule Slime.Parser do
   defp parse_attributes(line) do
     delim = String.first(line)
     if Dict.has_key?(@attr_list_delims , delim) do
-      line = String.replace_prefix(line, delim, "")
+      line = replace_prefix(line, delim, "")
```
Comment (Rakoth) [inline lib/slime/parser.ex]: How about String.slice(line, 1..-1)

**Comment by Rakoth** [comment]
👍 @doomspork any thoughts before merge?

**Comment by doomspork** [comment]
@Rakoth no sir, this looks good to me 👍 

Thank you @bozydar!!

**Comment by Rakoth** [comment]
@bozydar could you squash commits before merge?

**Comment by bozydar** [comment]
@Rakoth , @doomspork  I've already squashed the commits. Thank you for your review and advice!

**Pull request merged**